source/g4collimation/CollimationParticleGun.cpp

```diff
@@ -66,13 +69,21 @@ void CollimationParticleGun::SetParticleDetails(double x, double y, double px, d
 		}
 
 		//Remove states other than the ground state otherwise we get a crash.
+		bool e_state = false;
 		while(pdgid%10 !=0)
 		{
+			std::cout << "GEANT4> WARNING: Non-ground state particle!: " << pdgid << std::endl;
 			pdgid--;
+			e_state = true;
 		}
-
 		particle = G4IonTable::GetIonTable()->GetIon(pdgid);
-
+		if(e_state)
+		{
+			//rescale mass + energy here
+			const G4double mp = particle->GetPDGMass();
+			e = (e/mass)*mp;
```

Comment:
probably we could add a WARNING on the G4 STDOUT is the energy correction (i.e. `abs(mp/mass-1)`) is greater than a certain threshold, e.g. 1E-6